CMakeLists.txt

```diff
@@ -199,9 +216,16 @@ if(JERRY_LIBC)
   jerry_add_link_flags(-nostdlib)
 endif()
 
+if(USING_MSVC)
+  jerry_add_link_flags(/OPT:NOREF)
+  # Disable msvc  warning 4996 warning globally because it's stop us to use standard
+  # c function, refer to https://msdn.microsoft.com/en-us/library/ttcz0bys.aspx
```

Comment:
I'd suggest some grammar fixes even though I'm not a native speaker myself:

Disable MSVC warning 4996 globally because it stops us from using standard C functions

BTW, I've read the linked doc and found:
- "C4996 occurs when the compiler encounters a function or variable that is marked as deprecated. ... You can turn this warning off with the warning pragma or the /wd4996 command line option.": If that's the case, then adding the deprecated declspec in -compiler.h will not be really useful.
- "Some POSIX functions in the CRT have been renamed to conform with C99 and C++03 rules": I've always thought that we are C99 conforming. What are the problematic functions?

Reply:
That's the nasty thing of MSVC, such as `fopen` would raise warning.
it's would recommend you to use fopen_s.
And there is a lot more, such as strcpy_s and so on.
basically it's about the standard  ISO/IEC TR24731
http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1225.pdf